pkgs/servers/sql/cockroachdb/default.nix

```
@@ -1,70 +1,12 @@
-{ lib, stdenv, buildGoPackage, fetchurl
-, cmake, xz, which, autoconf
-, ncurses6, libedit, libunwind
-, installShellFiles
-, removeReferencesTo, go
-}:
-
-let
-  darwinDeps = [ libunwind libedit ];
-  linuxDeps  = [ ncurses6 ];
-
-  buildInputs = if stdenv.isDarwin then darwinDeps else linuxDeps;
-  nativeBuildInputs = [ installShellFiles cmake xz which autoconf ];
-
-in
-buildGoPackage rec {
-  pname = "cockroach";
-  version = "20.1.8";
-
-  goPackagePath = "github.com/cockroachdb/cockroach";
-
-  src = fetchurl {
-    url = "https://binaries.cockroachdb.com/cockroach-v${version}.src.tgz";
-    sha256 = "0mm3hfr778c7djza8gr1clwa8wca4d3ldh9hlg80avw4x664y5zi";
+self: {
+  cockroachdb_20_1 = self.lib.warn "Upstream support for v20.1 ends on 2021-11-12, please upgrade" (self.callPackage ./gopkg.nix {});
```

```suggestion
  cockroachdb_20_1 = self.lib.warn ''
    Upstream support for Cockroach v20.1 ends on 2021-11-12.
    Please upgrade by using the `cockroachdb_20_2` or `cockroachdb_21_1` package,
    or prepare to be automatically updated as the old version will be removed after that date.
  '' (self.callPackage ./gopkg.nix {});
```

I guess we can make the message a bit more descriptive so that people know what to do

---

Is that sufficient?

---

Sounds good :+1: